core/src/main/scala/zio/jdbc/ZConnectionPoolConfig.scala

```diff
@@ -32,7 +32,8 @@ object ZConnectionPoolConfig {
 
   lazy val default: ZConnectionPoolConfig = ZConnectionPoolConfig(8, 32, defaultRetryPolicy, 300.seconds)
 
-  lazy val defaultRetryPolicy: Schedule.WithState[Long, Any, Any, Duration] = Schedule.exponential(10.millis)
+  lazy val defaultRetryPolicy: Schedule.WithState[(Long, Long), Any, Any, (Long, Duration)] =
+    Schedule.recurs(10) && Schedule.exponential(15.millis)
```

Comment:
Why `10` and `15.millis` here? Because if my math is correct, it'll make the user wait for max 30 seconds.

See https://scastie.scala-lang.org/HnnTNtrRRaGlrsE8cxbajA